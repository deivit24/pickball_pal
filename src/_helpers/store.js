import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
