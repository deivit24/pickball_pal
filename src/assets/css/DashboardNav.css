#full-width-tabpanel-3 .MuiBox-root {
  padding: 10px 0px;
}
