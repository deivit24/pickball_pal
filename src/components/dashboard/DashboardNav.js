// React
import React, { useEffect, useState } from 'react';
//Prop Types
import PropTypes from 'prop-types';
//Redux
import { useSelector, useDispatch } from 'react-redux';
// Dashboard and Coversation components
import DashboardProfile from './DashboardProfile';
import DashboardPlaces from './DashboardPlaces';
import DashboardPosts from './DashboardPosts';
import Conversation from '../messages/Conversation';
import { makeStyles } from '@material-ui/core/styles';
/*
This is a componente straight from 
Material UI not much has changed
*/
import { AppBar, Tabs, Tab, Typography, Box } from '@material-ui/core';
// UUID
import { v4 as uuidv4 } from 'uuid';
// Get current profile and Get messages action
import { getCurrentProfile, getMessages } from '../../_actions/profile';

import '../../assets/css/DashboardNav.css';
function TabPanel(props) {
  const { children, value, index, ...other } = props;

  return (
    <Typography
      component="div"
      role="tabpanel"
      hidden={value !== index}
      id={`full-width-tabpanel-${index}`}
      aria-labelledby={`full-width-tab-${index}`}
      {...other}
    >
      {value === index && <Box p={3}>{children}</Box>}
    </Typography>
  );
}

TabPanel.propTypes = {
  children: PropTypes.node,
  index: PropTypes.any.isRequired,
  value: PropTypes.any.isRequired,
};

function a11yProps(index) {
  return {
    id: `full-width-tab-${index}`,
    'aria-controls': `full-width-tabpanel-${index}`,
  };
}

//Leaving the useStyles here
const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.background.paper,
    width: '100%',
  },
}));

const DashboardNav = () => {
  const meProfile = useSelector((store) => store.profile);
  const dispatch = useDispatch();
  const { messages } = meProfile;

  useEffect(() => {
    dispatch(getCurrentProfile());
    dispatch(getMessages());
  }, [dispatch]);
  const classes = useStyles();

  const [value, setValue] = useState(0);

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  let dispalyMessages =
    messages.length === 0 ? (
      <p>No messages</p>
    ) : (
      messages.map((el) => (
        <div key={uuidv4()}>
          <Conversation conversation={el} />
        </div>
      ))
    );

  return (
    <div className={classes.root}>
      <AppBar position="static" color="transparent">
        <Tabs
          value={value}
          onChange={handleChange}
          indicatorColor="primary"
          textColor="primary"
          variant="scrollable"
          scrollButtons="auto"
          aria-label="full width tabs example"
        >
          <Tab label="Bio" {...a11yProps(0)} />
          <Tab label="Places" {...a11yProps(1)} />
          <Tab label="Messages" {...a11yProps(2)} />
          <Tab label="Manage Posts" {...a11yProps(3)} />
        </Tabs>
      </AppBar>

      <TabPanel value={value} index={0}>
        <DashboardProfile />
      </TabPanel>
      <TabPanel value={value} index={1}>
        <DashboardPlaces />
      </TabPanel>
      <TabPanel value={value} index={2}>
        {dispalyMessages}
      </TabPanel>
      <TabPanel value={value} index={3}>
        <DashboardPosts />
      </TabPanel>
    </div>
  );
};

export default DashboardNav;
