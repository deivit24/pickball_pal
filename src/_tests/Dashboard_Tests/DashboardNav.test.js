import React from 'react';
import { render } from '@testing-library/react';
import DashboardNav from '../../components/dashboard/DashboardNav';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing DashboardNav

describe('Testing the DashboardNav', () => {
  it('it renders DashboardNav without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <DashboardNav />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <DashboardNav />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
