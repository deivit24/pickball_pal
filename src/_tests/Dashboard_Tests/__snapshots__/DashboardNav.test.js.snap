// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the DashboardNav matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-7"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorTransparent MuiPaper-elevation4"
    >
      <div
        class="MuiTabs-root"
      >
        <div
          class="MuiTabs-scrollable"
          style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
        />
        <div
          class="MuiTabs-scroller MuiTabs-scrollable"
          style="margin-bottom: 0px;"
        >
          <div
            aria-label="full width tabs example"
            class="MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-controls="full-width-tabpanel-0"
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
              id="full-width-tab-0"
              role="tab"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                Bio
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-controls="full-width-tabpanel-1"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
              id="full-width-tab-1"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                Places
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-controls="full-width-tabpanel-2"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
              id="full-width-tab-2"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                Messages
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-controls="full-width-tabpanel-3"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
              id="full-width-tab-3"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                Manage Posts
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="PrivateTabIndicator-root-8 PrivateTabIndicator-colorPrimary-9 MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </header>
    <div
      aria-labelledby="full-width-tab-0"
      class="MuiTypography-root MuiTypography-body1"
      id="full-width-tabpanel-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-12"
      >
        <div
          class="container Loading"
        >
          <div
            style="fill: #3f50b5; height: 100px; width: 100px;"
          >
            <svg
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              
  
              <path
                d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
                opacity=".25"
              />
              
  
              <path
                d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z"
              >
                
    
                <animatetransform
                  attributeName="transform"
                  dur="0.8s"
                  from="0 16 16"
                  repeatCount="indefinite"
                  to="360 16 16"
                  type="rotate"
                />
                
  
              </path>
              

            </svg>
            

          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="full-width-tab-1"
      class="MuiTypography-root MuiTypography-body1"
      hidden=""
      id="full-width-tabpanel-1"
      role="tabpanel"
    />
    <div
      aria-labelledby="full-width-tab-2"
      class="MuiTypography-root MuiTypography-body1"
      hidden=""
      id="full-width-tabpanel-2"
      role="tabpanel"
    />
    <div
      aria-labelledby="full-width-tab-3"
      class="MuiTypography-root MuiTypography-body1"
      hidden=""
      id="full-width-tabpanel-3"
      role="tabpanel"
    />
  </div>
</DocumentFragment>
`;
